tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -312,6 +404,26 @@ protected:
          InvokeReadCallbacks(to);
    }
 
+   /// General implementation of bulk read. Loop over the required range and read values that are required
+   /// and not already present. Derived classes may implement more optimized versions of this method.
+   /// See ReadBulk() for the return value.
+   virtual std::size_t ReadBulkImpl(const RBulkSpec &bulkSpec);
+
+   /// Returns the number of newly available values, that is the number of bools in bulkSpec.fMaskAvail that
+   /// flipped from false to true. As a special return value, kAllSet can be used if all values are read
+   /// independent from the masks.
+   std::size_t ReadBulk(const RBulkSpec &bulkSpec)
+   {
+      if (fIsSimple) {
+         /// For simple types, ignore the mask and memcopy the values into the destination
+         fPrincipalColumn->ReadV(bulkSpec.fFirstIndex, bulkSpec.fCount, bulkSpec.fValues);
+         memset(bulkSpec.fMaskAvail, 1, bulkSpec.fCount);
```

Comment:
nitpick (I almost didn't add this comment to the review but hey, I typed it, it's now yours to do whatever you want with it): this relies on the fact that `sizeof(bool)` is and will always be 1. I guess if that ever becomes false (as it's not actually guaranteed by the standard) there is _a lot more code_ that will have to be changed, but if you want to be a but more future-proof and declarative:

```suggestion
         std::fill(bulkSpec.fMaskAvail, bulkSpec.fMaskAvail + bulkSpec.fCount, true);
```

It should compile down to a `memset` anyway.